extensions/amp-sidebar/1.0/component.js

```diff
@@ -238,14 +292,16 @@ export function SidebarToolbar({
   }, [mediaQuery, toolbarTarget, targetEl]);
 
   return (
-    <nav
-      ref={ref}
-      toolbar={mediaQueryProp}
-      toolbar-target={toolbarTargetProp}
-      {...rest}
-    >
-      {children}
-    </nav>
+    preactMode && (
```

Comment:
Ideally the Preact component should not be concerned with the mode it is run in. Let's separate more distinctly: 

- Move `AmpSidebarToolbar` into `base-element.js` as it is only necessary for the custom element integration. Consider renaming to `ToolbarShim` or similar since the custom element version may not be "AMP".
- It seems like the context is used for two purposes:
  - `preactMode` tells us whether or not to render 
  - `domElement` gives us local references for cloning, `document` and `window` if we choose not to render
  Given this, can we ask for more generic props that would both be useful to the AMP layer and have sensible defaults? 
 
WDYT about doing something like the following?
```jsx
// component.js
const DEFAULT_GET_DOCUMENT = (element) => element?.ownerDocument;
const GET_CLONE = (element) => element.cloneNode(true);

SidebarToolbar({shouldRender = true, getDocument = DEFAULT_GET_DOCUMENT, getToolbarTargetElement = GET_CLONE}) {
  // Sanitize target
  useEffect(() => {
    const win = getDocument(ref?.current)?.defaultView;
    ...
  }, [getDocument])

  // Sanitize media query
  useEffect(() => {
    const win = getDocument(ref?.current)?.defaultView;
    ...
  }, [getDocument])

  // Append media query style
  useEffect(() => {
    const nodeToInsertIntoTarget = getToolbarTargetElement(ref?.current);
    ...
  }, [getDocument])

  return (shouldRender && <nav>)
}
```
```jsx
// base-element.js
ToolbarShim() {
  return <SidebarToolbar shouldRender={false} getDocument={() => domElement.ownerDocument} getToolbarTargetElement={() => domElement.cloneNode(true)} />
}
```

Another option could be to create a media query specific hook, and have both `SidebarToolbar` use it and render DOM, and the AMP layer could use it without rendering DOM:
```jsx
// src/preact/utils.js or similar
function useMediaQuery({target, getToolbarTargetElement, mediaQuery}) {
  // Sanitize media query
  // Append media query style
  useEffect(() => {
    ...
  }, [mediaQuery, getToolbarTargetElement, target])
}
```

```jsx
// component.js
function SidebarToolbar({toolbarTarget, toolbar: mediaQueryProp}) {
  // Sanitize target
  // Use media query hook
  useMediaQuery(targetEl, () => ref.current?.cloneNode(true), toolbar)
  return <nav>
}
```
```jsx
// base-element.js
function ToolbarShim({toolbarTarget, toolbar: mediaQueryProp}) {
  // Sanitize target
  // Use media query hook
  useMediaQuery(targetEl, () => domElement.cloneNode(true), toolbar)
  return null;
}
```

Reply:
Thanks!  Refactored into a custom hook and moved Amp component into `base-element` renaming as `ToolbarShim`.